core/shared/src/main/scala/fs2/Pull.scala

```diff
@@ -48,20 +123,85 @@ import Pull._
   *
   * `raiseError` is caught by `handleErrorWith`:
   *   - `handleErrorWith(raiseError(e))(f) == f(e)`
+  *
+  * @tparam F[_] the type of functional effects that can be performed by this pull.
+  * @tparam O The type of outputs emitted by this Pull.
+  * @tparam R The type of carry-over result that is returned by this Pull, if it returns successfully.
   */
 sealed abstract class Pull[+F[_], +O, +R] {
```

Comment:
The methods on `Pull` were roughly in alphabetical order, with a handful of exception cases. This PR reorders them a bit, though I don't understand the scheme.

Reply:
The scheme I followed was to group the functor-level methods, `map`, `as`, and `void`, and to bring forward the monadic errors `flatMap` and `>>`, which are the most heavily used ones.

Reply:
OK, I don't feel strongly about this as the Pull API is pretty small. Maybe we could add some comment fencing like `// === Functor operations ===`, not sure.